### DISCUSSION TITLE
Toast.Make() 'Title' parameter for use with WinUI

### DISCUSSION THREAD

**Discussion by dg2k** [CommunityToolkit/Maui#644] [General]
I am migrating Xamarin code to .NET MAUI and using the **CommunityToolkit.Maui**.

I understand that the message 'Title' parameter is missing for the cross-platform:
`Toast.Make(string message, ToastDuration duration = ToastDuration.Short, double textSize = AlertDefaults.FontSize)`.

@VladislavAntonyuk -- Is there any easy way to introduce Title parameter for WinUI?

**Comment by VladislavAntonyuk** [comment]
You need to use Snackbar. Toast doesn't have Title (It comes from Android implementation)

**Reply by dg2k** [reply]
Thank you @VladislavAntonyuk -- that's what I thought based on my initial research.

The tricky thing about Snackbar is that the `DisplaySnackbar` extension requires a VisualElement to anchor to.
What is the recommended **general purpose** VisualElement that can be used in most scenarios, simply to show the equivalent of a Toast message?

I was thinking of `Application.Current.MainPage` but not sure.   OK this seems to work but your expert opinion still required.

Wait, ... I thought the Title parameter is there ... where do I define the Title?

**Reply by VladislavAntonyuk** [reply]
You can use static method Snackbar.Make(...).Show
Or using MainPage as an anchor should also work

**Reply by dg2k** [reply]
@VladislavAntonyuk -- thanks, yes, both work. 

But where do I supply the Title?

**Reply by VladislavAntonyuk** [reply]
Sorry, I was wrong. We use only Text:

https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs#L19

**Reply by VladislavAntonyuk** [reply]
Title and Message are the same and the only field that exists in Toast and Snackbar. If you need a Title and Message for WinUI, you need to implement your own Snackbar/Toast

**Comment by CliffAgius** [comment]
Closed and answered
